### ISSUE TITLE
nesting does not  work

### ISSUE THREAD

**Issue by akostadinov** [deivid-rodriguez/pry-byebug#71]
I see that in vanilla `pry` if I have `binding.pry` on some place, it works as a break point. With `pry-byebug` these are ignored. Is there any way to not ignore them? I see that setting breakpoints dynamically is possible. But how to put static break points?

**Comment by deivid-rodriguez** [comment]
Hei! So there's two different issues here.

For the first one, you're right, `pry-byebug` cannot stop at breakpoints when code is evaluated from command prompt. I might revisit this in the future, but currently "subdebuggers" are not allowed.

For the second one, `pry` automatically requires plugins by default even if you specifiy `require: false` in your Gemfile (I know, weird). But it allows the user to configure that behaviour through the `Pry.config.should_load_plugins = false` configuration. Just drop it in your `.pryrc` file and you'll get a more sane behaviour!

**Comment by akostadinov** [comment]
Ok, thanks, .pryrc that can help have both options.

I'm just interested, what is the reason to not allow sub-debuggers? Or is it just current implementation. If it is a question of preference, maybe it needs to be done optional?

**Comment by deivid-rodriguez** [comment]
Yes, it's just a limitation of the current implementation, I'm not against it.

**Comment by JESii** [comment]
Hi... I've run into the same problem and it's breaking my normal workflow of entering pry and calling various routines with binding.prys to see how things are working. Any chance sub-debuggers can be added? I'd be willing to help if I can. Thanks.

**Comment by deivid-rodriguez** [comment]
Do you guys would like a proper `pry-byebug` console (with debugging commands) or a regular `pry` would be enough. If you want to the latter, I think stop monkeypatching `pry` and create "our own `binding.pry`" with debugging capabilities would work.

Proper debugging inside REPL evaluation commands needs to be fixed/added in `byebug` itself, either by default or through a custom command as suggested in deivid-rodriguez/byebug#121

**Comment by deivid-rodriguez** [comment]
I pushed a fix to this. Could you please try `pry-byebug` + latest `byebug` master and tell me if it works?

**Comment by akostadinov** [comment]
Thank you, do you have nightly builds or should I compile from source?

**Comment by deivid-rodriguez** [comment]
The easiest way to try this is, if you're using `bundler` is:

```
git clone https://github.com/deivid-rodriguez/pry-byebug
git clone https://github.com/deivid-rodriguez/byebug
cd byebug
bundle
bundle exec rake compile
```

and then point to your local clones in the Gemfile of your app:

```
gem 'pry-byebug', path: '/path/to/your/local/pry-byebug/clone'
gem 'byebug', path: '/path/to/your/local/byebug/clone'
```

**Comment by JESii** [comment]
Thanks! I'll try to give this a shot today...

**Comment by JESii** [comment]
Didn't work for me; here's what I did after building the latest versions and updating my Gemfile (the bundle install confirmed that I was using my local source).
1) added binding.pry to one of my methods
2) Executed a cucumber step which simply drops me into pry with a binding.pry
3) Called the method in step 1 above
4) The method completed without executing the contained binding.pry.
Did I miss something?

**Comment by deivid-rodriguez** [comment]
Your steps seem correct, I'll have another look. Could you confirm at least this works using `byebug` only?

**Comment by JESii** [comment]
OK; let me check...

**Comment by JESii** [comment]
Works just fine with byebug only...

**Comment by deivid-rodriguez** [comment]
:+1:

**Comment by JESii** [comment]
Hi, Deivid... Is this ready to be updated to our Gemfile for both pry-byebug and byebug or is there something else you need me to test? Thanks.

**Comment by deivid-rodriguez** [comment]
Nope, this needs more work.

**Comment by JESii** [comment]
OK; thanks... let me know if there's anything you need me to do.
